test/unit/org/apache/cassandra/db/compaction/unified/ControllerTest.java

```diff
@@ -270,7 +270,7 @@ public void testGetNumShards_growth_0()
         assertEquals(3 * (int) Controller.MAX_SHARD_SPLIT, controller.getNumShards(Math.scalb(10, 60)));
         assertEquals(3 * (int) Controller.MAX_SHARD_SPLIT, controller.getNumShards(Double.POSITIVE_INFINITY));
         // Check NaN
-        assertEquals(3, controller.getNumShards(Double.NaN));
+        assertEquals(1, controller.getNumShards(Double.NaN));
```

Comment:
Why this change?

Reply:
Because of the switch to go with no sharding for unknown flush size (in that case the flush size is returned as `Double.NaN`), see the change in `Controller.java`.